### PR TITLE
Views

### PR DESCRIPTION
**Pull request by aradradev**
Pull Request: In this project, I have implemented the following features:

- Implement the design from the sneak peek wireframes.
- Make sure that all views look very nice and closer to the wireframes and design

Please review this PR, and give your feedback if necessary.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,2 +1,27 @@
...
+    </div>
+    <div class="mt-8">
+      <h2 class="text-xl font-semibold mb-4">Recent Posts</h2>
+      <ul class="space-y-4">
+        <% @user.recent_posts.each do |post| %>
+          <li>
+            <a href="<%= post_path(post) %>" class="text-blue-500 hover:underline"><%= post.title %></a>
+          </li>
+        <% end %>
+      </ul>
```
Comment (danmainah) [inline app/views/users/show.html.erb]: - [ ] @aradradev Great work on your index show page, however, the page is supposed to display some parts of post's text and also each post's comments and likes counter.


Given Design          |  Your design     
:-------------------------:|:-------------------------:
image | image

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,2 +1,14 @@
-<h1>All Posts</h1>
-<p>This is a placeholder for posts index content.</p>
+<ul class="space-y-4">
```
Comment (danmainah) [inline app/views/posts/index.html.erb]: - [ ] @aradradev Great work  on your posts page, however the page is supposed to have details about the author and the number of posts at the top of the page, check the given design below to understand it better, also the pagination button is missing
image

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,14 @@
...
+    <li class="bg-white shadow-md p-4 rounded-lg">
+      <h2 class="text-2xl font-semibold mb-2">
+        <%= link_to post.title, post, class: "text-blue-500 hover:underline" %>
+      </h2>
+      <p class="text-gray-700 mb-2"><%= post.text %></p>
+      <p class="text-gray-600">Author: <%= post.author.name %></p>
+      <p class="text-gray-600">Comments: <%= post.comments_counter %></p>
+      <p class="text-gray-600">Likes: <%= post.likes_counter %></p>
+    </li>
+  <% end %>
```
Comment (danmainah) [inline app/views/posts/index.html.erb]: - [ ] @aradradev Great work listing all the user's posts, however as shown in the initial design below each post is supposed to display the five most recent comments of the post, kindly consider adding it.
image

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1,5 +1,21 @@
...
+      <div class="mb-4">
+        <p class="font-semibold">
+          <%= comment.user.name %>
+        </p>
+        <p>
+          <%= comment.text %>
+        </p>
+      </div>
+    <% end %>
+  </div>
```
Comment (danmainah) [inline app/views/posts/show.html.erb]: - [ ] @aradradev Great work displaying the user post with all its comments, however you forgot to add the comments and likes counter for this page.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @aradradev , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Implemented the functionality in actions methods for the posts user likes and comments controllers to enable display of saved data.
✔️ Added views for the required four pages following the design given.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- Kindly consider switching from using SQLite database to using the recommended PostgreSQL database, this will enable you to familiarise yourself with using Relational database which will be a great skill even in the job market.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by TadesseDev — APPROVED** [review]
### Hi @aradradev 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**